src/Containers/packaging/build/Microsoft.NET.Build.Containers.targets

```
@@ -11,7 +11,8 @@
                                  OR $(NETCoreSdkVersion.Contains('-')) == false
                                 )
                           )">true</_IsSDKContainerAllowedVersion>
-    <_ContainerIsTargetingNet8TFM>$([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '8.0'))</_ContainerIsTargetingNet8TFM>
+    <_ContainerIsTargetingNet8TFM>false</_ContainerIsTargetingNet8TFM>
+    <_ContainerIsTargetingNet8TFM Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND $([MSBuild]::VersionGreaterThanOrEquals('$(TargetFrameworkVersion)', '8.0'))">true</_ContainerIsTargetingNet8TFM>
```

This can be more succinctly represented:

```suggestion
    <_ContainerIsTargetingNet8TFM Condition="$([MSBuild]::IsTargetFrameworkCompatible($(TargetFramework), 'net8.0'))">true</_ContainerIsTargetingNet8TFM>
```

BUT that will fail when `$(TargetFramework)` is undefined, as it is in the currently-failing case in the outer build of a multitargeting project.

I think the best fix is to change the imports so that the container-publishing stuff is not imported for outer builds.